cohort-engine-api-web/src/test/java/com/ibm/cohort/engine/api/service/vt/DefaultVT.java

```diff
@@ -219,9 +262,70 @@ public void testXMLUnitAssertXPathValues() throws Exception {
         assertXpathValuesNotEqual("//jvm[2]/@current", "//jvm[3]/@current", myJavaFlavours);
     }
 
+    @Test
+	public void testValueSetUpload(){
+		final String RESOURCE = getUrlBase() + "/{version}/valueset";
+
+		String auth = dataServerConfig.getUser() + ":" + dataServerConfig.getPassword();
+		byte[] encodedAuth = Base64.encodeBase64(auth.getBytes(StandardCharsets.ISO_8859_1));
+
+		RequestSpecification request = buildBaseRequest(new Headers(new Header("Authorization", new String(encodedAuth))))
+				.param("version", ServiceBuildConstants.DATE)
+				.queryParam("fhir_server_rest_endpoint", dataServerConfig.getEndpoint())
+				.queryParam("updateIfExists", false)
+				.multiPart(CohortEngineRestHandler.VALUE_SET_PART, new File("src/test/resources/2.16.840.1.113762.1.4.1114.7.xlsx"));
+
+		ValidatableResponse response = request.post(RESOURCE, getServiceVersion()).then();
+		runSuccessValidation(response, ContentType.JSON, HttpStatus.SC_OK);
+	}
+
+	@Test
+	public void testValueSetAlreadyExists(){
+		testValueSetUpload();
```

Comment:
Does this rely on being run after the previous testValueSetUpload() method in order to be successful? If so, consider rewriting the test so that it is completely self sufficient. For example, you could call the upload twice in this method. Test order is not guaranteed by the test runners.

Reply:
The name of the upload is misleading, but the testValueSetUpload does not have the test annotation and does not run on its own. The two tests that depend on it run that method, which both uploads and confirms that the response is good, and the cleanup method deletes the value set each time. I would be fine with either rewording the upload method or combining the three tests into one, I just wanted to be clear that there currently isn't an order dependency.

Reply:
I realize looking back on this that I highlighted the part of the code that actually answered my question. You are doing it right. No action needed.